### PR TITLE
app-editors/qhexedit2, dev-embedded/ponyprog: version bump

### PR DESCRIPTION
**Pull request by bearsh**
_app-editors/qhexedit2_ is currently python3_6 only and uses a forked version which is behind upstream. _dev-embedded/ponyprog_, which needs a version bump is AFAIK the only consumer of that fork in tree. it doesn't seem _ponyprog_'s modification will land in upstream _qhexedit2_ soon. furthermore _ponyprog_ simply integrates its fork in its build system.

proposed solution: switch _qhexedit2_ to upstream project, use forked version of _qhexedit2_ only for _ponyprog_.

### CODE REVIEW

**Comment by juippis** [comment]
@ktrace could you check and give your opinions?

File: app-editors/qhexedit2/qhexedit2-0.8.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline app-editors/qhexedit2/qhexedit2-0.8.6.ebuild]: Use the latest available EAPI. 7 should work.

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: Same here.

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+inherit cmake-utils
```
Comment (juippis) [inline dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: Switch to cmake.eclass.

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+src_unpack() {
+	default
+	mv qhexedit2-*/src ${P}/qhexedit2/ || die "moving qhexedit2 failed"
+}
+
+src_configure() {
+	local mycmakeargs=( -DCMAKE_INSTALL_DOCDIR="share/doc/${PF}" )
+	cmake-utils_src_configure
```
Comment (juippis) [inline dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: (Remember to update this)

**Comment by ktrace** [comment]
Lancos sync his fork of qhexedit with upstream so now it looks like some lancos patches on top of qhexedit2 by simsys. As I understand I need fix my PR (right now it broken completely) to simsys for totally merge lancos's changes. So I prefer one lib qhexedit2 in portage and two ebuilds depend on it: sqlitebrowser and ponyprog. So this PR may be used as temporally solution.

File: app-editors/qhexedit2/qhexedit2-0.8.6.ebuild (hunk 1 above)
Comment (bearsh) [reply app-editors/qhexedit2/qhexedit2-0.8.6.ebuild]: done

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild (hunk 2 above)
Comment (bearsh) [reply dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: :ok:

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild (hunk 3 above)
Comment (bearsh) [reply dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: :ok:

File: dev-embedded/ponyprog/ponyprog-3.1.1.ebuild (hunk 4 above)
Comment (bearsh) [reply dev-embedded/ponyprog/ponyprog-3.1.1.ebuild]: :ok:

**Comment by ktrace** [comment]
Correct PR to upstream ready:
https://github.com/Simsys/qhexedit2/pull/107

**Comment by bearsh** [comment]
as @ZeroChaos- bumped qhexedit2 and I'm not really interested in ponyprog I'm gonna close this...

**Pull request closed without merging**